rule/package-comments.go

```diff
@@ -31,9 +31,48 @@ func (*PackageCommentsRule) Apply(file *lint.File, _ lint.Arguments) []lint.Fail
 	fileAst := file.AST
 	w := &lintPackageComments{fileAst, file, onFailure}
 	ast.Walk(w, fileAst)
+	failures = append(failures, checkPackageComment(file.Pkg)...)
 	return failures
 }
 
+var checkPackageCommentCache = map[*lint.Package]struct{}{}
+
+func checkPackageComment(p *lint.Package) []lint.Failure {
+	// deduplicate warnings in package
+	if _, ok := checkPackageCommentCache[p]; ok {
+		return nil
+	}
+	checkPackageCommentCache[p] = struct{}{}
+	var docFile *ast.File
+	var firstFile *ast.File
+	var firstFileName string
+	for name, file := range p.Files() {
+		if file.AST.Doc != nil {
+			return nil
+		}
+		if name == "doc.go" {
+			docFile = file.AST
+		}
+		if firstFileName == "" || firstFileName > name {
+			firstFile = file.AST
+			firstFileName = name
+		}
+	}
+	// prefer warning on doc.go
+	if docFile == nil {
+		docFile = firstFile
+	}
+	if docFile != nil {
+		return []lint.Failure{{
+			Category:   "comments",
+			Node:       docFile,
+			Confidence: 1,
+			Failure:    "should have a package comment, unless it's in another file for this package",
```

Comment:
the failure's message is not very _confident_: it can be interpreted as "there is a problem **unless** there is not" 
Creating a failure with confidence = 1 and a not so confident message seems to be somewhat contradictory

Reply:
maybe we can adjust message to `should have a package comment`.
the implementation should be confidence now.